### PR TITLE
bump credo version

### PR DESCRIPTION
**Pull request by ayrat555**
fixes https://github.com/poanetwork/blockscout/issues/2426

## Changelog

- bump credo version

Note:
I fixed all refactoring `with clause` warnings
and created an issue to fix duplicated code warnings (https://github.com/poanetwork/blockscout/issues/2436). I disable this check for now

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/controllers/chain/market_history_chart_controller.ex [hunk 1]
```diff
@@ -5,26 +5,28 @@ defmodule BlockScoutWeb.Chain.MarketHistoryChartController do
...
-        end
-
-      json(conn, %{
-        history_data: market_history_data,
-        supply_data: available_supply(Chain.supply_for_days(), exchange_rate)
-      })
-    else
-      _ -> unprocessable_entity(conn)
+    case ajax?(conn) do
+      true ->
```
Comment (pasqu4le) [inline apps/block_scout_web/lib/block_scout_web/controllers/chain/market_history_chart_controller.ex]: Does `credo` specifically asks to use `case` here?

This change could be smaller by using an `if` (and keeping the `else`) given that `ajax?` returns a boolean.

Same goes in other 2 places for the same `ajax?` function

File: apps/block_scout_web/lib/block_scout_web/controllers/recent_transactions_controller.ex [hunk 2]
```diff
@@ -6,30 +6,32 @@ defmodule BlockScoutWeb.RecentTransactionsController do
...
-        Chain.recent_collated_transactions(
-          necessity_by_association: %{
-            :block => :required,
-            [created_contract_address: :names] => :optional,
-            [from_address: :names] => :optional,
-            [to_address: :names] => :optional
-          },
-          paging_options: %PagingOptions{page_size: 5}
-        )
+    case ajax?(conn) do
```
Comment (Heimdell) [inline apps/block_scout_web/lib/block_scout_web/controllers/recent_transactions_controller.ex]: That can be an `if` as well.

File: apps/indexer/lib/indexer/fetcher/internal_transaction.ex [hunk 3]
```diff
@@ -231,6 +185,57 @@ defmodule Indexer.Fetcher.InternalTransaction do
...
+
+    case Chain.import(%{
+           addresses: %{params: addresses_params},
+           internal_transactions: %{params: internal_transactions_params_without_failed_creations},
+           internal_transactions_indexed_at_blocks: %{
+             params: internal_transactions_indexed_at_blocks,
+             with: :number_only_changeset
+           },
+           timeout: :infinity
+         }) do
```
Comment (Heimdell) [inline apps/indexer/lib/indexer/fetcher/internal_transaction.ex]: I recommend assigning expression like this to a name `imports = ...` and then using that name `case imports do ...`.

**Comment by ayrat555** [comment]
@vbaranov fixed

**Comment by vbaranov** [comment]
@ayrat555 dialyzer test is still failing

**Comment by ayrat555** [comment]
@vbaranov typespec was wrong. I fixed it

**Pull request merged**